rust/connlib/tunnel/src/tests/sim_client.rs

```diff
@@ -710,8 +710,12 @@ impl RefClient {
             return None;
         }
 
-        self.cidr_resource_by_ip(query.dns_server.ip())
-            .or(self.internet_resource)
+        let maybe_active_cidr_resource = self.cidr_resource_by_ip(query.dns_server.ip());
+        let maybe_active_internet_resource = self
+            .internet_resource
+            .filter(|r| !self.disabled_resources.contains(r));
```

Comment:
@conectado There was another bug here where we didn't check whether the Internet resource got disabled.